lnwallet/aux_signer.go

```diff
@@ -0,0 +1,146 @@
+package lnwallet
+
+import (
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/fn"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+// BaseAuxJob is a struct that contains the common fields that are shared among
+// the aux sign/verify jobs.
+type BaseAuxJob struct {
+	// OutputIndex is the output index of the HTLC on the commitment
+	// transaction being signed.
+	OutputIndex int32
```

Comment:
this should be unsigned int rather than signed?

Reply:
We take the value comes from here: https://github.com/lightningnetwork/lnd/blob/a350ccd7/lnwallet/channel.go#L308
So it can actually be negative. Added the same comment here.